cmd/kubeadm/app/phases/controlplane/manifests_linux_test.go

```diff
@@ -0,0 +1,217 @@
+			},
+		},
+		Groups: &users.EntryMap{
+			Entries: map[string]*users.Entry{
+				kubeadmconstants.KubeAPIServerUserName: {
+					ID: runAsGroup,
+				},
+				kubeadmconstants.ServiceAccountKeyReadersGroupName: {
+					ID: supGroup,
+				},
+			},
+		},
+	}
+
+	updatedFiles := map[string]ownerAndPermissions{}
+
+	if err := runKubeAPIServerAsNonRoot(&pod, usersAndGroups, cfg, func(path string, uid, gid int64, perms uint32) error {
+		updatedFiles[path] = ownerAndPermissions{UID: uid, GID: gid, Permissions: perms}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	verifyPodSecurityContext(t, &pod, runAsUser, runAsGroup, []int64{supGroup})
+
+	verifyContainerSecurityContext(t, pod.Spec.Containers[0], []v1.Capability{"NET_BIND_SERVICE"}, []v1.Capability{"ALL"}, nil)
+
+	wantUpdateFiles := map[string]ownerAndPermissions{
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPublicKeyName):   {UID: runAsUser, GID: runAsGroup, Permissions: 0600},
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPrivateKeyName):  {UID: 0, GID: supGroup, Permissions: 0640},
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKeyName):              {UID: runAsUser, GID: runAsGroup, Permissions: 0600},
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKubeletClientKeyName): {UID: runAsUser, GID: runAsGroup, Permissions: 0600},
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyClientKeyName):       {UID: runAsUser, GID: runAsGroup, Permissions: 0600},
+		filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerEtcdClientKeyName):    {UID: runAsUser, GID: runAsGroup, Permissions: 0600},
+	}
+
+	verifyFilePermissions(t, updatedFiles, wantUpdateFiles)
+}
+
+func TestRunKubeControllerManagerAsNonRoot(t *testing.T) {
+	cfg := &kubeadm.ClusterConfiguration{}
+	endpoint := kubeadm.APIEndpoint{}
+	pod := GetStaticPodSpecs(cfg, &endpoint)[kubeadmconstants.KubeControllerManager]
+	var runAsUser, runAsGroup, supGroup int64 = 1000, 1001, 1002
+
+	usersAndGroups := &users.UsersAndGroups{
+		Users: &users.EntryMap{
+			Entries: map[string]*users.Entry{
+				kubeadmconstants.KubeControllerManagerUserName: {
+					ID: runAsUser,
+				},
+			},
+		},
+		Groups: &users.EntryMap{
+			Entries: map[string]*users.Entry{
+				kubeadmconstants.KubeControllerManagerUserName: {
+					ID: runAsGroup,
+				},
+				kubeadmconstants.ServiceAccountKeyReadersGroupName: {
+					ID: supGroup,
+				},
+			},
+		},
+	}
+
+	updatedFiles := map[string]ownerAndPermissions{}
+
+	if err := runKubeControllerManagerAsNonRoot(&pod, usersAndGroups, cfg, func(path string, uid, gid int64, perms uint32) error {
+		updatedFiles[path] = ownerAndPermissions{UID: uid, GID: gid, Permissions: perms}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	verifyPodSecurityContext(t, &pod, runAsUser, runAsGroup, []int64{supGroup})
+
+	verifyContainerSecurityContext(t, pod.Spec.Containers[0], nil, []v1.Capability{"ALL"}, pointer.Bool(false))
```

Comment:
we can omit some of the empty lines.

prior `}` provide sufficient separation for readability.

Reply:
Removed empty lines. Thanks!